src/shogun/util/factory.h

```diff
@@ -96,16 +96,16 @@ namespace shogun
 	}
 
 	CFeatures* string_features(
-	    CFile* file, EAlphabet alpha = DNA,
-	    EPrimitiveType primitive_type = PT_CHAR)
+	    CFile* file, machine_int_t alphabet_type = DNA,
```

Comment:
doesn't swig take care of this automatically? I.e. is this strictly necessary?

Reply:
It doesn't compile differently. The other factory methods use the same types I used.